test/hummingbot/connector/test_connector_metrics_collector.py

```diff
@@ -0,0 +1,289 @@
+import asyncio
+import json
+import platform
+from copy import deepcopy
+from decimal import Decimal
+from typing import Awaitable
+from unittest import TestCase
+from unittest.mock import MagicMock, PropertyMock
+
+import hummingbot.connector.connector_metrics_collector
+from hummingbot.client.config.config_var import ConfigVar
+from hummingbot.client.config.global_config_map import global_config_map
+from hummingbot.connector.connector_metrics_collector import DummyMetricsCollector, TradeVolumeMetricCollector
+from hummingbot.core.data_type.common import TradeType, OrderType
+from hummingbot.core.data_type.trade_fee import AddedToCostTradeFee
+from hummingbot.core.event.events import OrderFilledEvent, MarketEvent
+from hummingbot.core.rate_oracle.rate_oracle import RateOracle
+
+
+class TradeVolumeMetricCollectorTests(TestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self._global_config_backup = deepcopy(global_config_map)
+
+        self.metrics_collector_url = "localhost"
+        self.connector_name = "test_connector"
+        self.instance_id = "test_instance_id"
+        self.client_version = "0.1"
+        self.rate_oracle = RateOracle()
+        self.connector_mock = MagicMock()
+        type(self.connector_mock).name = PropertyMock(return_value=self.connector_name)
+        self.dispatcher_mock = MagicMock()
+        type(self.dispatcher_mock).log_server_url = PropertyMock(return_value=self.metrics_collector_url)
+
+        self.original_client_version = hummingbot.connector.connector_metrics_collector.CLIENT_VERSION
+        hummingbot.connector.connector_metrics_collector.CLIENT_VERSION = self.client_version
+
+        self.metrics_collector = TradeVolumeMetricCollector(
+            connector=self.connector_mock,
+            activation_interval=10,
+            metrics_dispatcher=self.dispatcher_mock,
+            rate_provider=self.rate_oracle,
+            instance_id=self.instance_id,
+            client_version=self.client_version)
+
+    def tearDown(self) -> None:
+        hummingbot.connector.connector_metrics_collector.CLIENT_VERSION = self.original_client_version
+        self._reset_global_config()
+        super().tearDown()
+
+    def _reset_global_config(self):
+        for key, value in self._global_config_backup.items():
```

Comment:
I just realized that this should prob start with `global_config_map.clear()` to remove any new key-values added